controllers/chat/sendSignedMessage.js

```diff
@@ -0,0 +1,79 @@
+const Validator = require('fastest-validator');
+const {StreamChat} = require('stream-chat');
+const {responseError, responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+
+const v = new Validator();
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const sendSignedMesage = async (req, res) => {
+  const schema = {
+    channelId: 'string|empty:false',
+    message: 'string|empty:false',
+    channelType: {
+      type: 'enum',
+      values: Object.values(CHANNEL_TYPE),
+      empty: false
+    }
+  };
+  const validated = v.validate(req.body, schema);
+  if (validated.length)
+    return res.status(403).json({
+      message: 'Error validation',
+      error: validated
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The validation logic is well implemented using the `fastest-validator` library. However, it would be better to move this validation schema and logic into a separate middleware function. This way, the validation can be reused in other parts of the application if needed, and it also keeps the controller function clean and focused on its main task.

```diff
-  const schema = {
-    channelId: 'string|empty:false',
-    message: 'string|empty:false',
-    channelType: {
-      type: 'enum',
-      values: Object.values(CHANNEL_TYPE),
-      empty: false
-    }
-  };
-  const validated = v.validate(req.body, schema);
-  if (validated.length)
-    return res.status(403).json({
-      message: 'Error validation',
-      error: validated
-    });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->